### ISSUE TITLE
Provide positional information in errors dictionary when serializing a collection (many=True)

### ISSUE THREAD

**Issue by sloria** [marshmallow-code/marshmallow#75]
When validating a collection with invalid elements, the returned errors dictionary does not provide any information about the positions of invalid elements

Example:

``` python
def validate_age(age):
    if age < 21:
        raise ValidationError('Must be over 21')

class UserSchema(Schema):
    age = fields.Number(validate=validate_age)

in_data = [{'age': 24}, {'age': 19}]
data, errors = UserSchema(many=True).load(in_data)
```
### Current

``` python
errors  # => {'age': ['Must be over 21']}
```
### Possible solution

Provide positional information in a nested dictionary, keyed by index.

``` python
errors  # => {'age': {1: ['Must be over 21']}}
```

**Comment by sloria** [comment]
An alternative format to the one proposed above would be to key the errors dictionary by the indices of invalid items:

``` python
errors  # => {1: {'age': ['Must be over 21']}}
```

I've implemented this behavior in a09fa92, though it's not finalized. This feature is still up for discussion.

Another thought: We could add a `class Meta`  option to turn this error indexing behavior on or off. Perhaps we default it to be turned off and release it in the next feature release (1.3) so that it can get some use before finalizing it in 2.0.

**Comment by sloria** [comment]
Implemented in https://github.com/marshmallow-code/marshmallow/commit/a3d36024bfd6b76ecc0d0c031a5c93ea8400a90d
